src/components/Reports/useSDGProgress.ts

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from 'react';
+import useDebounce from 'src/utils/useDebounce';
+import { Report, SDG } from 'src/types/Report';
+
+const DEBOUNCE_TIME_MS = 500;
+
+export default function useSDGProgress(
```

Comment:
This is a big ugly hook which adds state and debounce for all of the Sustainable Development Goal Progress fields

Reply:
Made it much less big and less ugly. Thanks @karthikbtf!

Reply:
awesome, thanks @charlie-tf !
can we iterate over `0..16` and call this with the index? or does react want a more deterministic order of hooks?

Reply:
I initially tried putting this into a map expression, then a for-loop. React didn't like either. I think deterministically calling each is what it wants here.

Reply:
Got it.. fine as-is. Thanks 👍